src/main/java/hudson/plugins/analysis/core/BuildHistory.java

```diff
@@ -187,11 +190,26 @@ private boolean isSuccessfulAction(final ResultAction<? extends BuildResult> act
         return null;
     }
 
-    private boolean hasValidResult(final AbstractBuild<?, ?> build) {
+    /**
+     * Added for backward compatibility. It generates <pre>AbstractBuild getReferenceBuild()</pre> bytecode during the build
+     * process, so old implementations can use that signature.
+     * 
+     * @see {@link WithBridgeMethods}
+     */
+    @Deprecated
+    private Object getReferenceAbstractBuild(Run run, Class targetClass) {
+        if (run != null && run instanceof AbstractBuild) {
```

Comment:
Are you suggesting to remove ```run != null```? I'd be ok for me, but I think it's clearer like it is now.